components/salsa-macro-rules/src/setup_input_struct.rs

```diff
@@ -53,6 +53,12 @@ macro_rules! setup_input_struct {
         // The function used to implement `C::heap_size`.
         heap_size_fn: $($heap_size_fn:path)?,
 
+        // If `true`, a `serde_fn` has been provided.
+        serializable: $serializable:tt,
```

Comment:
Should we call this persistable? `serializable` suggests that it only indicates whether a type as `serialize` but not necessarily `deserialize`.